applications: nrf_desktop: Update memory layout documentation (#29532)
nrfconnect/sdk-nrf: applications/nrf_desktop/doc/dfu.rst

Configuration
*************

To perform the firmware upgrade, you must enable the bootloader.
You can use the DFU module with either MCUboot or B0 bootloader.
For more information on how to enable and configure a bootloader, see the :ref:`nrf_desktop_bootloader` section.

> Apart from proper bootloader configuration, you need to also specify memory layout.

Reviewer: ```suggestion
In addition to proper bootloader configuration, you also need to specify the memory layout.
```
